doc/en/developer/source/source.rst

```diff
@@ -166,30 +169,112 @@ Each branch has the following structure::
 Git client configuration
 ------------------------
 
+Git client maintains configuration for global settings, system settings, and local settings.
+
+To review local settings:
+
+.. code-block:: bash
+   
+   $ git config --local --get-regexp core.*
+   
+    core.repositoryformatversion 0
+    core.filemode true
+    core.bare false
+    core.logallrefupdates true
+    core.ignorecase true
+    core.precomposeunicode false
+
+To review global settings:
+
+.. code-block:: bash
+
+   $ git config --global --get-regexp core.*
+
+   core.autocrlf input
+   core.safecrlf true
+   core.ignorecase true
+
+Line endings
+^^^^^^^^^^^^
+
 When a repository is shared across different platforms it is necessary to have a 
 strategy in place for dealing with file line endings. In general git is pretty good about
 dealing this without explicit configuration but to be safe developers should set the 
-``core.autocrlf`` setting to "input"::
+``core.autocrlf`` setting to "input":
 
-    % git config --global core.autocrlf input
+.. code-block:: bash
+
+   $ git config --global core.autocrlf input
 
 The value "input" essentially tells git to respect whatever line ending form is present
 in the git repository.
 
 .. note::
 
    It is also a good idea, especially for Windows users, to set the ``core.safecrlf`` 
-   option to "true"::
+   option to "true":
 
-      % git config --global core.safecrlf true
+   .. code-block:: bash
+   
+      $ git config --global core.safecrlf true
 
    This will basically prevent commits that may potentially modify file line endings.
 
 Some useful reading on this subject:
 
-* http://www.kernel.org/pub/software/scm/git/docs/git-config.html
-* https://help.github.com/articles/dealing-with-line-endings
-* http://stackoverflow.com/questions/170961/whats-the-best-crlf-handling-strategy-with-git
+* `git config <https://git-scm.com/docs/git-config>`__ (git)
+* `Configuring Git to handle line endings <https://help.github.com/articles/dealing-with-line-endings>`__ (GitHub)
+* `What's the best CRLF (carriage return, line feed) handling strategy with Git? <http://stackoverflow.com/questions/170961/whats-the-best-crlf-handling-strategy-with-git>`__ (Stack Overflow)
+
+Unicode filenames
+^^^^^^^^^^^^^^^^^
+
+Occasionally their are two different representations of a unicode character:
+
++-------------------------+---------------+------------------+
+| Representation          | Example       | Operating System |
++=========================+===============+==================+
+| Precomposed form        | ``Ü``         | Linux, Windows   |
++-------------------------+---------------+------------------+
+| Decomposed form         | ``U`` + ``¨`` | macOS            |
++-------------------------+---------------+------------------+
+
+Care must be taken by macOS configure the git client appropriately to committing files in decomposed form.
```

Comment:
Care must be taken by macOS users to configure the git client appropriately for committing files in decomposed form.